Dockerfile

```diff
@@ -386,9 +386,6 @@ RUN --mount=type=cache,sharing=locked,id=moby-dev-aptlib,target=/var/lib/apt \
             dbus-user-session \
             systemd \
             systemd-sysv
-RUN mkdir -p hack \
-  && curl -o hack/dind-systemd https://raw.githubusercontent.com/AkihiroSuda/containerized-systemd/b70bac0daeea120456764248164c21684ade7d0d/docker-entrypoint.sh \
-  && chmod +x hack/dind-systemd
```

Comment:
./cc @AkihiroSuda (FYI - this commit adds a copy/variation of the above in `hack/`)

(I recall there originally was some discussion about this in https://github.com/moby/moby/pull/40493#discussion_r383409927)

Reply:
Copying the script SGTM, but what is the modification here?